src/flow.jl

```diff
@@ -1102,6 +1235,236 @@ function update(sw::ShallowWaterLand{T}, swr::ShallowWaterRiver{T}, network, Δt
+"""
+function flow_area(profile::FloodPlainProfile{T}, h, i::Int, i1::Int, i2::Int)::T where {T}
+    Δh = h - profile.depth[i1]
+    a = profile.a[i][i1] + profile.width[i][i2] * Δh
+    return a
+end
+
+"""
+    function wetted_perimeter(profile::FloodPlainProfile{T}, h, i::Int, i1::Int)
+
+Compute floodplain wetted perimeter based on flow depth `h`, floodplain `profile` at
+index 'i' and floodplain depth `profile` index `i1`.
+"""
+function wetted_perimeter(profile::FloodPlainProfile{T}, h, i::Int, i1::Int)::T where {T}
+    Δh = h - profile.depth[i1]
+    p = profile.p[i][i1] + 2.0 * Δh
+    return p
+end
+
+"Compute flood depth by interpolating flood volume `flood_volume` using flood depth intervals."
+function flood_depth(profile::FloodPlainProfile{T}, flood_volume, dl, i::Int)::T where {T}
+    i1, i2 = interpolation_indices(flood_volume, profile.volume[i])
+    ΔA = (flood_volume - profile.volume[i][i1]) / dl
+    Δh = ΔA / profile.width[i][i2]
+    flood_depth = profile.depth[i1] + Δh
+    return flood_depth
+end
+
+"Initialize floodplain geometry and `FloodPlain` parameters"
+function initialize_floodplain_1d(
+    nc,
+    config,
+    inds,
+    riverwidth,
+    riverlength,
+    index_pit,
+    n_edges,
+    nodes_at_link,
+)
+
+    n_floodplain = ncread(
+        nc,
+        config,
+        "lateral.river.floodplain.n";
+        sel = inds,
+        defaults = 0.072,
+        type = Float,
+    )
+    volume = ncread(
+        nc,
+        config,
+        "lateral.river.floodplain.volume";
+        sel = inds,
+        type = Float,
+        dimname = :flood_depth,
+    )
+    n = length(inds)
+
+    # for convenience (interpolation) flood depth 0.0 m is added, with associated area (a),
+    # volume, width (river width) and wetted perimeter (p).
+    volume = vcat(fill(Float(0), n)', volume)
+    start_volume = volume
+    flood_depths = Float.(nc["flood_depth"][:])
+    pushfirst!(flood_depths, 0.0)
+    n_depths = length(flood_depths)
+
+    p = zeros(Float, n_depths, n)
+    a = zeros(Float, n_depths, n)
+    segment_volume = zeros(Float, n_depths, n)
+    width = zeros(Float, n_depths, n)
+    width[1, :] = riverwidth[1:n]
+
+    # determine flow area (a), width and wetted perimeter (p)
+    h = diff(flood_depths)
+    incorrect_vol = 0
+    riv_cells = 0
+    error_vol = 0
+    for i = 1:n
+        riv_cell = 0
+        diff_volume = diff(volume[:, i])
+
+        for j = 1:(n_depths-1)
+            # assume rectangular shape of flood depth segment
+            width[j+1, i] = diff_volume[j] / (h[j] * riverlength[i])
+            # check provided flood volume (floodplain width should be constant or increasing
+            # as a function of flood depth)
+            if width[j+1, i] < width[j, i]
+                incorrect_vol += 1
+                riv_cell = 1
+                error_vol =
+                    error_vol + ((width[j, i] - width[j+1, i]) * h[j] * riverlength[i])
```

Comment:
After fixing this in https://github.com/Deltares/hydromt_wflow/pull/123 I'm still getting a warning here. The diff in width is however very small and I suspect it's a rounding error. These rounding errors might be important for the calculation, but warning message like the one below might also confuse users. 

> The provided volume of 734 rectangular floodplain schematization segments for 132 river cells (6.63 % of total river cells) is not correct and has been increased with 0.0 % of provided volume.

I suggest to only raise a warning if the error is relative large, for instance by allowing for a 1cm diff in width:

```suggestion
                # raise warning only if diff is larger than rounding error of 1 cm
                if (width[j, i] - width[j+1, i]) > 0.01
                    incorrect_vol += 1
                    riv_cell = 1
                    error_vol =
                        error_vol + ((width[j, i] - width[j+1, i]) * h[j] * riverlength[i])
                end
```

Reply:
Good suggestion! Changed this in commit https://github.com/Deltares/Wflow.jl/pull/233/commits/0a4ffa88e0123fd4c48947f967cc2c377cb1e233.

Reply:
With this suggested change for the Rhine model % of total river cells not correct dropped from 41.82 to 0.46, while the increase in total volume (correction) stayed the same (0.18 % of provided volume). For the Rhine model there is also a mismatch between river width used in hydromt_wflow and in Wflow I think (`wflow_riverwidth_sobek_global`), explaining the required correction. In addition @JoostBuitink did some flood profile testing, and also concluded that mismatches were mostly caused by rounding errors.